src/platform_impl/windows/monitor.rs

```diff
@@ -162,6 +163,220 @@ impl MonitorHandle {
+            ) == ERROR_SUCCESS
+        } {
+            continue;
+        }
+
+        let got = String::from_utf16_lossy(&drv_buf[..(drv_len as usize / 2)]);
+
+        if !got.starts_with(inst) {
+            continue;
+        }
+
+        // Open Device Parameters
+        let params = format!("{subkey}\\Device Parameters");
+        let params_w = to_wide(&params);
+        let mut hkey3: ScopeGuard<HKEY, _> = guard(std::ptr::null_mut(), |v| unsafe {
+            RegCloseKey(v);
+        });
+
+        if unsafe {
+            RegOpenKeyExW(HKEY_LOCAL_MACHINE, params_w.as_ptr(), 0, KEY_READ, &mut *hkey3)
+                != ERROR_SUCCESS
+        } {
+            continue;
+        }
+
+        let edid_w = to_wide("EDID");
+        let mut edid = [0u8; 256];
+        let mut edid_len = edid.len() as u32;
+        if unsafe {
+            RegQueryValueExW(
+                *hkey3,
+                edid_w.as_ptr(),
+                null_mut(),
+                null_mut(),
+                edid.as_mut_ptr(),
+                &mut edid_len,
+            ) != ERROR_SUCCESS
+        } || edid_len < 23
+        {
+            continue;
+        }
+        let width_mm: u32;
+        let height_mm: u32;
+
+        // We want to have more detailed resolution than centimeters,
+        // specifically millimeters. EDID provides Detailed Timing
+        // Descriptor (DTD) table which can contain desired
+        // size. There can be up to 4 DTDs in EDID, but the first one
+        // non-zero-clock DTD is the monitor’s preferred (native)
+        // mode, so we try only it.
+        const DTD0: usize = 54;
+
+        let pixel_clock = (edid[DTD0] as u16) | ((edid[DTD0 + 1] as u16) << 8);
+
+        if pixel_clock != 0 {
+            // For mm precision we need 12-14 bits from Detailed
+            // Timing Descriptor
+            // https://en.wikipedia.org/wiki/Extended_Display_Identification_Data#Detailed_Timing_Descriptor.
+            let h_size_lsb = edid[DTD0 + 12] as u16;
+            let v_size_lsb = edid[DTD0 + 13] as u16;
+            let size_msb = edid[DTD0 + 14] as u16;
+            let h_msb = (size_msb >> 4) & 0x0f;
+            let v_msb = size_msb & 0x0f;
+
+            width_mm = ((h_msb << 8) | h_size_lsb) as u32;
+            height_mm = ((v_msb << 8) | v_size_lsb) as u32;
+        } else {
+            let width_cm = edid[21] as u32;
+            let height_cm = edid[22] as u32;
+
+            width_mm = width_cm * 10;
+            height_mm = height_cm * 10;
+        }
+
+        return Some((width_mm, height_mm));
+    }
+
+    None
+}
+
+fn monitor_physical_size(mi: &MONITORINFOEXW) -> Option<(u32, u32)> {
+    unsafe {
+        // adapter_name is the wide-string in MONITORINFOEXW.szDevice
+        let adapter_name = mi.szDevice.as_ptr();
+        // find the matching DISPLAY_DEVICEW
+        let dd = get_monitor_device(adapter_name)?;
+        // read EDID
+        read_size_from_edid(&dd)
+    }
+}
```

Comment:
Uff. This suddenly got a lot more complex and error-prone, I not longer feel confident enough in my Windows abilities to review this.

@wareya, do you have any input here? For context, see https://github.com/rust-windowing/winit/pull/4212#discussion_r2073310949.

(Also, looking at prior art, it seems that [GLFW uses `GetDeviceCaps`](https://github.com/glfw/glfw/blob/e7ea71be039836da3a98cea55ae5569cb5eb885c/src/win32_monitor.c#L83-L96), and that neither SDL nor FLTK supports querying the physical size of the screen at all. So maybe it's actually fine for us to use `GetDeviceCaps`?)

Reply:
PS: I missed the email notification for this ping and only noticed it because I was poking around the issue tracker; if you ping me again and I don't respond, feel free to ping me on element/matrix too.